pontoon/terminology/models.py

```diff
@@ -0,0 +1,251 @@
+from __future__ import unicode_literals
+
+from collections import defaultdict
+import logging
+
+from bulk_update.helper import bulk_update
+from django.db import models
+from django.db import transaction
+
+from pontoon.base.models import (
+    Entity,
+    Locale,
+)
+
+from pontoon.base.utils import (
+    get_words,
+    get_singulars
+)
+from pontoon.terminology.term_indexes import DBTermIndex
+
+log = logging.getLogger(__name__)
+
+
+class TermManager(models.Manager):
+    term_index = DBTermIndex
+
+    def find_terms(self, original_string, entity_singulars):
```

Comment:
I think this should be a simple function in this module. Maybe something else, I'm not entirely sure. But I strongly believe it shouldn't be a method of this Manager class.